src/keybindings.c

```diff
@@ -1827,11 +1827,11 @@ static void goto_matching_brace(GeanyDocument *doc)
 
 static gboolean cb_func_clipboard_action(guint key_id)
 {
-	GeanyDocument *doc = document_get_current();
+	GtkWidget *focusw = gtk_window_get_focus(GTK_WINDOW(main_widgets.window));
 
-	if (doc == NULL)
+	if (!IS_SCINTILLA(focusw)) {
```

Comment:
Works for me, e.g. in the toolbar search entry, or the scribble pane (just tested with paste, after setting it to a non-default key to be sure it wasn't the widget itself handling it)